presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixSqlTypeMapping.java

```diff
@@ -0,0 +1,419 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.phoenix;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.Session;
+import io.prestosql.spi.type.ArrayType;
+import io.prestosql.spi.type.TimeZoneKey;
+import io.prestosql.spi.type.VarbinaryType;
+import io.prestosql.tests.AbstractTestQueryFramework;
+import io.prestosql.tests.datatype.CreateAndInsertDataSetup;
+import io.prestosql.tests.datatype.CreateAsSelectDataSetup;
+import io.prestosql.tests.datatype.DataSetup;
+import io.prestosql.tests.datatype.DataType;
+import io.prestosql.tests.datatype.DataTypeTest;
+import io.prestosql.tests.sql.PrestoSqlExecutor;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.io.BaseEncoding.base16;
+import static io.prestosql.plugin.phoenix.PhoenixQueryRunner.createPhoenixQueryRunner;
+import static io.prestosql.spi.type.DateType.DATE;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.TimeZoneKey.UTC_KEY;
+import static io.prestosql.tests.datatype.DataType.bigintDataType;
+import static io.prestosql.tests.datatype.DataType.booleanDataType;
+import static io.prestosql.tests.datatype.DataType.dataType;
+import static io.prestosql.tests.datatype.DataType.dateDataType;
+import static io.prestosql.tests.datatype.DataType.doubleDataType;
+import static io.prestosql.tests.datatype.DataType.integerDataType;
+import static io.prestosql.tests.datatype.DataType.realDataType;
+import static io.prestosql.tests.datatype.DataType.smallintDataType;
+import static io.prestosql.tests.datatype.DataType.tinyintDataType;
+import static io.prestosql.tests.datatype.DataType.varbinaryDataType;
+import static io.prestosql.tests.datatype.DataType.varcharDataType;
+import static java.lang.String.format;
+import static java.math.RoundingMode.UNNECESSARY;
+import static java.nio.charset.StandardCharsets.UTF_16LE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toList;
+
+@Test
+public class TestPhoenixSqlTypeMapping
```

Comment:
We should try to de-duplicate these tests with `TestPostgreSqlTypeMapping`, but can figure that out later, as a future follow up.